### ISSUE TITLE
Make fwf_positions() guess end argument from start

### ISSUE THREAD

**Issue by rcragun** [tidyverse/readr#996]
I would expect `fwf_positions()` to guess the value of `end` based on the value of `start`. If `end` is not specified, the best guess should be that `end` is `lead(start)-1`. Perhaps I misunderstand what `arg=NULL` means, but having `end=NULL` implies to me that `end` is optional. I believe that this function could be improved with some simple changes:

```r
fwf_positions <- function (start, end, col_names = NULL) {
  if(missing(end)){
    end <- lead(start)-1L
  }
  col_names <- fwf_col_names(col_names, length(start))
  tibble(begin = start - 1L, end = end, col_names = col_names)
}
```

**Comment by jimhester** [comment]
`end = NULL` looks like it was inadvertantly added in https://github.com/tidyverse/readr/pull/616, it should probably just be removed.

**Comment by jimhester** [comment]
Fixed by e675e0a6
